mdpow/tests/test_automated_dihedral_analysis.py

```diff
@@ -0,0 +1,188 @@
+                                           'H23', 'H24', 'H25', 'H26', 'H27', 'H28', 'H29',
+                                           'H30', 'H31', 'H32', 'H33', 'H34', 'H35'], dtype=object)
+
+    check_hydrogens = np.array(['H21', 'H22', 'H23', 'H24', 'H25', 'H26', 'H27', 'H28',
+                                'H29', 'H30', 'H31', 'H32', 'H33', 'H34', 'H35'], dtype=object)
+
+    # pre-padding dihedral group statistics
+    DG_O1C2N3S4_mean = -0.13089566578887002
+    DG_O1C2N3S4_var = 0.03473127346296412
+
+    # pre-padding dihedral group statistics
+    DG_C13141520_mean = 90.04076626959608
+    DG_C13141520_var = 0.8814931303534448
+
+    # post-padding dihedral group statistics
+    # 'A' represents 'augmented'
+    ADG_C13141520_mean = 93.50126701923381
+    ADG_C13141520_var = 0.8815675813248334
+    
+    def test_build_universe(self, SM25_tmp_dir):
+        u = dihedrals.build_universe(dirname=SM25_tmp_dir)
+        solute = u.select_atoms('resname UNK')
+        solute_names = solute.atoms.names
+        assert solute_names.all() == self.universe_solute_atom_names.all()
+
+    def test_add_hydrogens(self, SM25_tmp_dir):
+        # need to finish building test dataset with no H for completion
+        # this test is correct once H removed from top and traj
+        # tests RDKit and add_hydrogens
+        u = dihedrals.build_universe(dirname=SM25_tmp_dir)
+        solute = dihedrals.rdkit_conversion(u, resname='UNK')[1]
+        solute_hydrogens = solute.select_atoms('name *H*').atoms.names
+        assert solute_hydrogens.all() == self.check_hydrogens.all()
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="pytest=7.2.0, build=py37h89c1867_0 gives wrong answers")
+    def test_dihedral_indices(self, gen_data):
+        atom_group_indices = gen_data[0]
+        assert atom_group_indices == self.check_atom_group_indices
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="pytest=7.2.0, build=py37h89c1867_0 gives wrong answers")
+    def test_SMARTS(self, gen_data):
+        atom_group_indices_alt = gen_data[1]
+        assert atom_group_indices_alt == self.check_atom_group_indices_alt
+
+    '''def test_dihedral_indices_error_exception(self, SM25_tmp_dir):
+        with pytest.raises(AttributeError) as excinfo:
+            ada.dihedral_indices(dirname=SM25_tmp_dir, resname=self.resname)
+        assert excinfo.value is AttributeError'''
+    # needs an example topology without hydrogen
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="pytest=7.2.0, build=py37h89c1867_0 gives wrong answers")
+    def test_dihedral_groups(self, SM25_tmp_dir):
+        groups = dihedrals.dihedral_groups(dirname=SM25_tmp_dir, resname=self.resname)
+        i = 0
+        while i < len(groups):
+            assert groups[i].all() == self.check_groups[i].all()
+            i+=1
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="scipy circvar gives wrong answers")
```

Comment:
This was one of the original testing issues we fixed for DihedralAnalysis, from which I copied over the `skipif` because `scipy.stats` is used in the same way here (caused the same failure). I suggested in an earlier comment that it might be pytest because of the similarity with it only being Python < 3.8. I will have to set up an environment to investigate and figure out if it is pytest or scipy.

Reply:
Any clearer about this? At minimum. add a comment that states what we know/suspect.

Reply:
Add issue number, add more text to be specific ... whoeever reads this has no clue what "nature of this problem" means.

Read the comments through the eyes of someone who is not an insider.